tests/Feature/RequestPaginationTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+use GuzzleHttp\Promise\PromiseInterface;
+use Illuminate\Support\LazyCollection;
+use Saloon\Contracts\Response;
+use Saloon\Tests\Fixtures\Connectors\OffsetRequestPaginatorConnector;
+use Saloon\Tests\Fixtures\Connectors\PageRequestPaginatorConnector;
+use Saloon\Tests\Fixtures\Requests\OffsetGetSuperHeroesRequest;
+use Saloon\Tests\Fixtures\Requests\PageGetSuperHeroesRequest;
+
+test('you can configure a page paginator, and iterate over every request/response', function (): void {
+    $connector = new PageRequestPaginatorConnector;
+    $request = new PageGetSuperHeroesRequest;
+
+    $paginator = $connector->paginate($request);
+
+    $responses = [];
+    $superheroes = [];
+
+    foreach ($paginator as $response) {
+        $responses[] = $response;
+        $superheroes = [...$superheroes, ...$response->json('data')];
+    }
+
+    expect($responses)->toHaveCount(4)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(20)->each->toBeArray();
+});
+
+test('you can configure a offset paginator, and iterate over every request/response', function (): void {
+    $connector = new OffsetRequestPaginatorConnector;
+    $request = new OffsetGetSuperHeroesRequest;
+
+    $paginator = $connector->paginate($request, limit: 5);
+
+    $responses = [];
+    $superheroes = [];
+
+    foreach ($paginator as $response) {
+        $responses[] = $response;
+        $superheroes = [...$superheroes, ...$response->json('data')];
+    }
+
+    expect($responses)->toHaveCount(4)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(20)->each->toBeArray();
```

Comment:
Again, the rest of this test is absolutely fine, but I would be certain what results you are getting back

Reply:
You could also check the pending requests for each response have the right query parameters too, that would be a great test assertion like

```php
expect($responses[0]->getPendingRequest()->query()->all())->toEqual(['page' => 1, 'per_page' => 5]);
expect($responses[1]->getPendingRequest()->query()->all())->toEqual(['page' => 2, 'per_page' => 5]);
expect($responses[2]->getPendingRequest()->query()->all())->toEqual(['page' => 3, 'per_page' => 5]);
expect($responses[3]->getPendingRequest()->query()->all())->toEqual(['page' => 4, 'per_page' => 5]);
```